### ISSUE TITLE
Strange campaign status / problem

### ISSUE THREAD

**Issue by jvanremoortere** [responsiv/campaign-plugin#9]
I've just sent a campaign out to 401 subscribers. For some reason it stopped after sending 259 emails leaving 142 emails unsent. The 259 mails that were sent all were delivered between 2016-02-22 21:25:03 and 2016-02-22 21:28:10.  The status keeps being "Active" and although the subscribers tab mentions the timestamp of the sent emails, the "sent" quantity in the stats keeps being 0. I checked Mandrill and I have an hour limit of 1471 emails so that couldn't be the case.

![schermafdruk 2016-02-22 23 50 54](https://cloud.githubusercontent.com/assets/4172138/13236051/3e6b4cea-d9c2-11e5-99d5-d0025ae3b70a.png)

**Comment by daftspunk** [comment]
My first guess is the 142 unsent emails were to users that exist as subscribers but have either unsubscribed or have not confirmed their address. If you click on a subscriber it should display a message about the status of their subscription, eg:

> This subscriber has not confirmed their subscription and will not receive messages.

**Comment by jvanremoortere** [comment]
That doesn't seem to be the case. I've just checked 10 random subscribers from the 142 and they all mention "This subscriber is confirmed and is ready to receive messages."

**Comment by daftspunk** [comment]
I'm not sure, your screenshot above says 0 messages were sent...

**Comment by jvanremoortere** [comment]
I know... I don't get it either. 0 sent, 12 opened (up till now 13 opened).

"-12 unread"? I don't know what that is about. Double negative = positive. So that could be right...

The sent ones do mention a timestamp. How does the sent items get calculated? Isn't this a count on the timestamp field of sent?

**Comment by daftspunk** [comment]
-12 indicates that 12 emails have been opened (and sent) but the plugin doesn't know about it. Perhaps there is an error occurring when sending mail. The logic goes like this:
1. Send mail to subscriber
2. Update the `sent_at` column for the campaign
3. Update the counter

So perhaps step 3 is not being reached. Please provide replication instructions using a fresh install. Also what sending provider are you using? How are you calling the cron job?

**Comment by jvanremoortere** [comment]
Mandrill mentions 259 emails were sent (same amount as campaign manager)

401 - 259 = 142 (without a sent timestamp)

I've just checked every subscriber who hasn't received an email manually and it seems like there were some faults in the imported email addresses (4 to be exact). Maybe it got stuck on one of them. Don't they get validated on import? It seems like they were never validated in the first place coming from the previous customer system...

One email address contained a "ü" for example... I've never seen it used before in an email address but I've just read that it could be possible? (damn you German speaking people). It just imported all the characters before the "ü".

I've just corrected all the faults manually. I will try to send the campaign again to check if the issue is fixed.
